File: FitTrack-Frontend/src/app/responses/user-profile/user-profile.response.ts
import { GenderEnum } from '../../enums/gender.enum';

export interface UserProfileResponse {
  username: string;
  email: string;
  firstName: string;
  lastName: string;
  dateOfBirth: Date;
  height: number;
  weight: number;
  gender: GenderEnum;
}

File: FitTrack-Backend/FitTrack.Data.Contract/Helpers/Responses/UserProfileResponse.cs
using FitTrack.Data.Object.Enums;

namespace FitTrack.Data.Contract.Helpers.Responses;
public class UserProfileResponse
{
    public string Username { get; set; }
    public string Email { get; set; }
    public string FirstName { get; set; }
    public string LastName { get; set; }
    public DateTime DateOfBirth { get; set; }
    public double Height { get; set; }
    public double Weight { get; set; }
    public GenderEnum Gender { get; set; }
}
